src/data/services/LmsApiService.js

```diff
@@ -0,0 +1,24 @@
+import qs from 'query-string';
+
+import apiClient from '../apiClient';
+
+class LmsApiService {
+  static UpdateEmailSettings(courseRunId, hasEmailsEnabled) {
+    const queryParams = {
+      course_id: courseRunId,
+      email_opt_in: hasEmailsEnabled,
+    };
+    const emailSettingsUrl = `${apiClient.authBaseUrl}/user_api/v1/preferences/email_opt_in/`;
+    return apiClient.post(
+      emailSettingsUrl,
+      qs.stringify(queryParams),
+      {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
Just to confirm, did you try making the API call without these headers? I think I remember when I tried making the API call locally when I was testing out the API initially, those headers were necessary. Just curious to see if you ran into that as well.

Reply:
I was getting a 400 Bad Request with a reason of "Missing Post Parameters"